<html>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "DTD/xhtml1-transitional.dtd">
<head>
	<title>Phylodynamic Datasets</title>
	<meta name = "keywords" content = "phylogenetics,phylogeography,phylodynamics,pathogens,livestock,sequences,epic,scotland" >
	<meta name = "description" content = "Academic Research and data in Virus and Bacteria Evolution and Modelling" >
	<meta http-equiv = "Content-Type" content = "text/html; charset = ISO-8859-1" >
	<LINK REL=StyleSheet HREF="style.css" TYPE="text/css" MEDIA=screen>	
</head>



	<div id = "page">
		<div id = "nav">
		
        <a href = "http://epicscotland.github.io/index.html">Home</a>
        <a href = "index.html">Phylodynamics</a>
        <a href = "data.html">Data</a>
        <a href = "links.html">Links</a>

		
		</div>
	

		<br>

		<div id = "content">
		

		<center><h1>Phylodynamic Datasets - Links</h1></center>
		
		<br>

		<h1>Databases</h1>

		<TABLE cellpadding=5 ALIGNMENT="top" border=1>
		<TR><TD WIDTH=250><b>Name<TD><b>Information</b></TR>
		<TR><TD><a href="http://www.ncbi.nlm.nih.gov/genbank/">GenBank</a><TD>NCBI Public database for all sequences</TR>
		<TR><TD><a href="http://www.ncbi.nlm.nih.gov/genomes/FLU/FLU.html">Influenza Virus Resource</a><TD>NCBI public database for influenza sequences (useful interface to GenBank)</TR>
		<TR><TD><a href="http://tree.bio.ed.ac.uk/rnavirusdb/">RNA Virus Database</a><TD>Database of RNA viruses, reference alignments and trees</TR>
		<TR><TD><a href="http://www.wrlfmd.org/">Foot-and-Mouth Disease</a><TD>FMD Database at Pirbright</TR>
		<TR><TD><a href="http://bioportal.ucdavis.edu/">Disease BioPortal</a><TD>FMD Portal for real time surveillance by University of California, Davis</TR>
		</TABLE>

		<h1>Software</h1>

		<TABLE cellpadding=5 ALIGNMENT="top" border=1>
		<TR><TD WIDTH=250><b>Name<TD><b>Information</b></TR>
		<TR><TD><a href="http://www.mbio.ncsu.edu/bioedit/bioedit.html">BioEdit</a><TD>Viewing and aligning sequences (Windows)</TR>
		<TR><TD><a href="http://www.megasoftware.net/">MEGA 5</a><TD>Generating trees, viewing and aligning sequences</TR>
		<TR><TD><a href="http://tree.bio.ed.ac.uk/software/pathogen/">Path-o-gen</a><TD>Test for molecular clock signal (e.g. before running BEAST)</TR>		
		<TR><TD><a href="http://beast.bio.ed.ac.uk/Main_Page">BEAST 1</a><TD>Generating time resolved bayesian tree sets with complex models</TR>
		<TR><TD><a href="http://www.beast2.org/wiki/index.php/Main_Page">BEAST 2</a><TD>Generating time resolved bayesian tree sets with complex models</TR>
		<TR><TD><a href="http://tree.bio.ed.ac.uk/software/tracer/">Tracer</a><TD>Analysing BEAST log file outputs, including mutation rate and other model paramter estimates</TR>
		<TR><TD><a href="http://tree.bio.ed.ac.uk/software/figtree/">FigTree</a><TD>Viewing and colouring phylogenetic trees</TR>
		</TABLE>
        
        <h1>Other resources</h1>
        
		<p>Institute of Veterinary Virology, University of Bern, useful BVDV website
		<a href="http://www.bvd-info.ch/veterinarians/virus/">BVD-info.ch</a>
		</p>

		<p>Institute of Virology, Hannover, Community Reference Laboratory for Classical Swine Fever Virus 
		<a href="http://viro08.tiho-hannover.de/eg/index.html">http://viro08.tiho-hannover.de/eg/index.html</a>
		</p>
        
        <h1>Scientific Papers</h1>
        
        <h2>ASFV</h2>
        <p>de Villier et al (2010) "Phylogenomic analysis of 11 complete African swine fever virus genome sequences" <i>Virology</i> <b>400</b>:128-136<br>
		<a href="http://dx.doi.org/10.1016/j.virol.2010.01.019">http://dx.doi.org/10.1016/j.virol.2010.01.019</a></p>
        
        <p>Lubisi et al (2005) "Molecular epidemiology of African swine fever in East Africa" <i>Archives of Virology</i> <b>150</b>:2439-2452<br>
		<a href="http://link.springer.com/article/10.1007%2Fs00705-005-0602-1">http://link.springer.com/article/10.1007%2Fs00705-005-0602-1</a></p>
        
        <p>Boshof et al (2007) "Genetic characterisation of African swine fever viruses from outbreaks in southern Africa (1973-1999)" <i>Veterniary Microbiology</i> <b>121</b>:45-55<br>
		<a href="http://dx.doi.org/10.1016/j.vetmic.2006.11.007">http://dx.doi.org/10.1016/j.vetmic.2006.11.007</a></p>
        
        <h2>BVDV</h2>
        
	<p>Richard E Booth, Carole J Thomas, Laila MR El-Attar, George Gunn and Joe Brownlie<br>
		"A phylogenetic analysis of Bovine Viral Diarrhoea Virus (BVDV) isolates from
		six different regions of the UK and links to animal movement data"<br>
		<i>Veterinary Research</i> 2013, <b>44:43</b>
		<a href="http://www.veterinaryresearch.org/content/44/1/43">Open access paper</a><br>
		
	<p>Chernick A, Godson DL, van der Meer F (2014) "Metadata beyond the sequence enables the phylodynamic inference of bovine viral diarrhea virus type 1a isolates from Western Canada"<br>
        <i>Infect Genet Evol</i>pii: S1567-1348(14)00006-9. doi: 10.1016/j.meegid.2014.01.003
        <a href="http://www.ncbi.nlm.nih.gov/pubmed/24424090">http://dx.doi.org/10.1016/j.meegid.2014.01.003</a>

	<h2>CSFV</h2>

	<p>Postel et al (2012) "Improved strategy for phylogenetic analysis of classical swine fever virus based on full-length E2 encoding sequences"<br> 
		<i>Vet Res.</i> 2012 Jun 7;43:50 <a href="http://www.ncbi.nlm.nih.gov/pubmed/22676246">doi: 10.1186/1297-9716-43-50</a><br>

	
        
		<br><br>
		
		</div>	

		<hr/>

		<div id = "footer">

			<TABLE>
			<TR>

			<TD WIDTH=100>
				<a href = "img/epiclogo_smaller.jpg">
				<img src = "img/epiclogo_smaller.jpg" title = "EPIC Scotland" alt = "EPIC Scotland" width=100>
				</a>
			</TD>

			<TD WIDTH=10></TD>
			<TD VALIGN="top">	Dr S. J. Lycett<br>
						Institute of Biodiversity, Animal Health and Comparative Medicine<br>
						University of Glasgow<br>
						Room 319, Jarrett Building, 464 Bearsden Road<br>
						Glasgow, G61 1QH<br>						
			</TABLE>


			<p id="s1">
				Copyright EPIC Scotland 2014
			</p>

		</div>

	</div>
</body>
</html>





